test/Infrastructure/LeanCode.Serialization.Tests/JsonConvertersTests.cs

```diff
@@ -35,6 +27,31 @@ public void Throws_when_attempting_to_deserialize_DateOnly_with_time_part(string
         Assert.Equal(reserialized, JsonSerializer.Serialize(date, LaxSerializerOptions));
     }
 
+    [Theory]
+    [InlineData("\"2021-12-15T07:23:10.115\"", 7, 23, 10, 115, "\"07:23:10.115\"")]
+    public void Correctly_deserializes_TimeOnly_with_date_part_when_using_lax_converter(
+        string serialized, int hour, int minute, int second, int millisecond, string reserialized)
+    {
+        var time = new TimeOnly(hour, minute, second, millisecond);
+
+        Assert.Equal(time, JsonSerializer.Deserialize<TimeOnly>(serialized, LaxSerializerOptions));
+        Assert.Equal(reserialized, JsonSerializer.Serialize(time, LaxSerializerOptions));
+    }
+
+    [Theory]
+    [InlineData("\"2021-12-15T07:23:10.115Z\"", 2021, 12, 15, 7, 23, 10, 115,1, "\"2021-12-15T07:23:10.115Z\"")]
```

Comment:
Let's add more cases that handle the "lax" part.

Also
```suggestion
    [InlineData("\"2021-12-15T07:23:10.115Z\"", 2021, 12, 15, 7, 23, 10, 115, 1, "\"2021-12-15T07:23:10.115Z\"")]
```